src/lib/variables/variables.ts

```diff
@@ -112,12 +111,12 @@ export const mergeVariables = (
   }
 
   return result;
-};
+}
 
-export const getValue = (
-  variables: NonPrimitive | undefined,
+export function getValue(
+  variables: NonPrimitive,
   key: string[]
-): Variables | undefined => {
+): Variables | undefined {
   if (variables === undefined) {
     return undefined;
   }
```

Comment:
suggestion: This should never happen (in types) - `NonPrimitive` is not allowed to be `undefined`. I think it can be removed.

Reply:
opinion: Also depending on where this is called maybe we should remove the try-catch below and let this function throw on errors so that we don't mix `undefined` as a valid value and `undefined` as a sentinel for error.

Reply:
Also also `Variables | undefined` is redundant because `Variables` already contains `None` which contains `undefined` - this is why I think we should enable the throwing behavior again - but that's a breaking change.

Reply:
You are right, that first condition is not needed, and I will remove it.

Regarding `getValue` removing all the things you correctly pointed, it looses reason to exists. Mainly because it is used in one place only (util's `replaceParameters`). So, I would move try-catch and getting value there.